stingray/bispectrum.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import division
+import numpy as np
+from scipy.linalg import hankel
+import scipy.fftpack
+import stingray.lightcurve as lightcurve
+import matplotlib.pyplot as plt
+
+# import stingray.utils as utils
+# from stingray.utils import simon
+
+__all__ = ["Bispectrum"]
+
+
+class Bispectrum(object):
+
+    def __init__(self, lc=None, m=64, nfft=128):
+        """
+        Parameters
+        ----------
+        lc: lightcurve.Lightcurve object, optional, default None
+        The light curve data to be Fourier-transformed.
+
+        m : Number of samples in each segment Default :64(expecting huge data)
+
+        nfft : Fft size Default is 128
+
+
+        attributes
+        ----------
+        k: Number of segments
+
+        m : Number of samples in each segment Default :64(expecting huge data)
+
+        nfft : Fft size Default is 128
+
+        nsamp  : total numberof samples from the input signal
+
+        """
+
+        if lc is not None:
+            pass
+        else:
+            self.k = None
+            self.nfft = nfft
+            self.m = m
+            self.nsamp = None
+            return
+        self.m = m
+        self.nfft = nfft
+        assert isinstance(lc, lightcurve.Lightcurve), \
+            "lc must be a lightcurve.Lightcurve object!"
+
+        lc.counts = lc.counts.ravel(order='C')
+        self.nsamp = lc.counts.shape[0]
+
+        assert isinstance(m, int), "m is not an int !"
+
+        assert isinstance(nfft, int), "nfft is not an int !"
+
+        self.k = self.nsamp // self.m
```

Comment:
I had to do this because bichorence wouldn't require compute_bispectra so if we include compute_bispectra in our __int_ we would dealing with extra unnecessary overhead( which can crash program due  to memory issues as we are dealing with huge data in my case working with 128 sized fft i could see considerable lagon my pc )( this was not the case in power spectra as every method of class had dependency on make_powerspectra) 
but there is an alternate 
remove bicoherence out of bipsectrum class ( but it  is a property of bispectrum so it wouldnt be aesthetic to  remove it out of class) 
So in all 
1.I could create a new class if you want me to for bicoherence 
2. or make changes you mentioned above (not considering memory issues)  @dhuppenkothen